lib/extensions/database_cleaner-activerecord-seeded_deletion.rb

```diff
@@ -0,0 +1,71 @@
+require 'database_cleaner-active_record'
+
+module DatabaseCleaner
+  module ActiveRecord
+    # SeededDeletion is a strategy that deletes all records from tables except those that existed before it was instantiated
+    # This is useful for tests that need the seeded data to be present.
+    class SeededDeletion < Deletion
+      def initialize(opts = {})
+        super
+        cache_max_seeded_ids
+      end
+
+      def self.table_max_id_cache
+        @table_max_id_cache ||= {}
+      end
+
+      # Memoize the maximum ID for each class table with non-zero number of rows
+      def self.table_max_id_cache=(table_id_hash)
+        @table_max_id_cache ||= table_id_hash
+      end
+
+      def cache_max_seeded_ids
+        self.class.table_max_id_cache = table_max_id_hash
+      end
+
+      delegate :table_max_id_cache, to: :class
+
+      private
+      def table_max_id_hash
+        h = {}
+        tables_to_clean(connection).each do |table_name|
+          begin
+            # Check if the table has an id column using SQL
+            column_exists_query = "SELECT EXISTS (
+              SELECT 1
+              FROM information_schema.columns
+              WHERE table_name = '#{table_name}'
+              AND column_name = 'id'
+            )"
+
+            has_id_column = connection.select_value(column_exists_query)
+
+            if has_id_column
+              # Get the maximum id using SQL
+              max_id = connection.select_value("SELECT MAX(id) FROM #{connection.quote_table_name(table_name)}")
```

Comment:
Currently, getting the max IDs when initializing the SeededDeletion object, so we can then ensure the cleanup will remove any newer IDs for each table... any table with no rows, will be skipped.

We can change this to use sequence but first wanted to get this here so it's reviewable.